server/src/main/scala/za/co/absa/atum/server/api/repository/PartitioningRepositoryImpl.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.repository
+
+import za.co.absa.atum.model.dto.{AdditionalDataSubmitDTO, PartitioningSubmitDTO}
+import za.co.absa.atum.server.api.database.runs.functions.{CreateOrUpdateAdditionalData, CreatePartitioningIfNotExists}
+import za.co.absa.atum.server.api.exception.DatabaseError
+import za.co.absa.fadb.exceptions.StatusException
+import zio._
+
+class PartitioningRepositoryImpl(
+  createPartitioningIfNotExistsFn: CreatePartitioningIfNotExists,
+  createOrUpdateAdditionalDataFn: CreateOrUpdateAdditionalData
+) extends PartitioningRepository with BaseRepository {
+
+  override def createPartitioningIfNotExists(
+    partitioning: PartitioningSubmitDTO
+  ): IO[DatabaseError, Either[StatusException, Unit]] = {
+    dbCallWithStatus(createPartitioningIfNotExistsFn(partitioning), "createPartitioningIfNotExists")
```

Comment:
These strings are used in Repositories as well as in Services. Please consider using constants.

Reply:
https://github.com/AbsaOSS/atum-service/issues/176